src/dotnet/APIView/APIViewWeb/Pages/Unauthorized.cshtml

```diff
@@ -24,7 +24,7 @@
     find yourself under the "People" tab of the organization page and set your visibility to public.
 </p>
 
-<a class="login-button btn btn-outline-dark" asp-action="Login" asp-controller="Account">
+<a class="login-button btn btn-outline-dark" asp-action="Login" asp-controller="Account" asp-route-returnUrl="@HttpContext.Request.Query["returnurl"]">
```

Comment:
Make the return URL a property on the page, binding directly to query feels a bit wrong.

Reply:
Updated